src/tasks/checkEnvironment.js

```diff
@@ -31,12 +36,17 @@ var self = {
 
     // check for windows or Mac OSX
     if (['win32', 'darwin'].indexOf(process.platform) > -1) {
-      return self.boot2dockerExists();
+      self.dockerMachine = self.dockerMachineExists();
+      if (!self.dockerMachine) {
+        self.boot2docker = self.boot2dockerExists();
+      }
+      return self.dockerMachine || self.boot2docker;
```

Comment:
I captured them in member variables, so that I didn't need to scan the disk again with `which` when I need to execute the `isDockerRunning` method.